builtin/providers/google/image.go

```diff
@@ -2,96 +2,186 @@ package google
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
+
+	"google.golang.org/api/googleapi"
 )
 
-// If the given name is a URL, return it.
-// If it is of the form project/name, search the specified project first, then
-// search image families in the specified project.
-// If it is of the form name then look in the configured project, then hosted
-// image projects, and lastly at image families in hosted image projects.
-func resolveImage(c *Config, name string) (string, error) {
+const (
+	resolveImageProjectRegex = "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?" // TODO(paddy): this isn't based on any documentation; we're just copying the image name restrictions. Need to follow up with @danawillow and/or @evandbrown and see if there's an actual limit to this
+	resolveImageFamilyRegex  = "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?" // TODO(paddy): this isn't based on any documentation; we're just copying the image name restrictions. Need to follow up with @danawillow and/or @evandbrown and see if there's an actual limit to this
```

Comment:
Just took a closer look and it's the same validation as images, so this looks like the right regex to use.